polymer.externs.js

```diff
@@ -1115,7 +1122,9 @@ var PolymerTouchEvent;
  * @typedef {{
  *   index: number,
  *   removed: !Array,
- *   addedCount: number
+ *   addedCount: number,
+ *   object: !Array,
```

Comment:
feel free to peruse the API[1] or code[2].

[1] http://polymer.github.io/polymer/
[2] https://github.com/Polymer/polymer/blob/1b02e96e77a0ef201976341fade763c617eca0d5/src/standard/notify-path.html#L418

typically you only `splice()` an `Array`.